### PR TITLE
Replaces search bar component with usememo logic

### PR DESCRIPTION
**Pull request by Amogh-Bharadwaj**
Split from #664

### CODE REVIEW

File: ui/app/mirrors/edit/[mirrorId]/cdc.tsx [hunk 1]
```diff
@@ -88,9 +88,15 @@ type SnapshotStatusProps = {
...
 };
 export const SnapshotStatusTable = ({ status }: SnapshotStatusProps) => {
-  const [snapshotRows, setSnapshotRows] = useState(
-    status.clones.map(summarizeTableClone)
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const snapshotRows = useMemo(
+    () =>
+      status.clones.map(summarizeTableClone).filter((row: any) => {
+        return row.tableName.toLowerCase().includes(searchQuery.toLowerCase());
+      }),
```
Comment (serprex) [inline ui/app/mirrors/edit/[mirrorId]/cdc.tsx:96]:
```suggestion
      status.clones.map(summarizeTableClone).filter((row: any) =>
        row.tableName.toLowerCase().includes(searchQuery.toLowerCase())
      ),
```

File: ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx [hunk 2]
```diff
@@ -53,13 +53,16 @@ const ROWS_PER_PAGE = 10;
...
   const endRow = startRow + ROWS_PER_PAGE;
-  const allRows = rows.slice(startRow, endRow);
-  const [displayedRows, setDisplayedRows] = useState(
-    rows.slice(startRow, endRow)
-  );
+  const displayedRows = useMemo(() => {
+    const allRows = rows.slice(startRow, endRow);
+    const shownRows = allRows.filter((row: any) => {
+      return row.batchId == parseInt(searchQuery, 10);
+    });
```
Comment (serprex) [inline ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx:63]:
```suggestion
    const shownRows = allRows.filter((row: any) =>
      row.batchId == parseInt(searchQuery, 10)
    );
```

File: ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx [hunk 3]
```diff
@@ -53,13 +53,16 @@ const ROWS_PER_PAGE = 10;
...
+    return shownRows.length > 0 ? shownRows : allRows;
```
Comment (serprex) [inline ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx:64]: Can include sort here

Comment (Amogh-Bharadwaj) [reply ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx:64]: I've branched this from main. Once this is merged I'll rebase https://github.com/PeerDB-io/peerdb/pull/664 accordingly

**Review by serprex — APPROVED** [review]

**Pull request merged**